### ISSUE TITLE
ckan/ckan-base:2.10.5, 2.10.6 fails with "uwsgi: unrecognized option: wsgi-file"

### ISSUE THREAD

**Issue by alexander-gesinn** [ckan/ckan-docker#199]
CKAN 2.10.5, 2.10.6 container fails to start with:
```
ckan-1        | uwsgi: unrecognized option: wsgi-file
ckan-1        | getopt_long() error
```
CKAN 2.11.1 works without any issues. Any idea what's going wrong?

**Comment by kowh-ai** [comment]
Do you get a `PermissionError: [Errno 13] Permission denied: '/srv/app/ckan.ini'` error too?

**Comment by nickramser** [comment]
I'm hitting this too. I'm not sure what changed, but the fix for me was to explicitly add `--plugins-dir /usr/lib/uwsgi/ --need-plugin python3,http` to the `UWSGI_OPTS` in `start_ckan.sh`. "unrecognized option: wsgi-file" is a symptom of the uwsgi-python plugin not being loaded.

**Comment by pauloneves** [comment]
Following @nickramser  orientation, ere is my working config:

```
UWSGI_OPTS="--plugins-dir /usr/lib/uwsgi/ \
            --need-plugin python3,http \
            --socket /tmp/uwsgi.sock \
            --wsgi-file /srv/app/wsgi.py \
            --module wsgi:application \
            --http 0.0.0.0:5000 \
            --master --enable-threads \
            --lazy-apps \
            -p 2 -L -b 32768 --vacuum \
            --harakiri $UWSGI_HARAKIRI"
```

I fixed it copying an changed start_ckan.sh to the docker "setup/start_ckan.sh.override" and adding this line to the docker file:

COPY setup/start_ckan.sh.override ${APP_DIR}/start_ckan.sh

**Comment by kowh-ai** [comment]
Are you guys able to use the **2.10.x Python/Debian** based images? These do not have this issue

Here is a snippet of the README

![Screenshot 2025-01-15 at 10 48 57 am](https://github.com/user-attachments/assets/0061a048-69c7-47a5-9503-6b4d42a0b50b)

**Comment by pauloneves** [comment]
You are right @kowh-ai ! Changing the base image to ckan/ckan-base:2.10.6-py3.10 and the wsgi problem was gone.

In my docker customization I had to change the user to root. It was weird, but worked. Must investigate more.

**Comment by alexander-gesinn** [comment]
@kowh-ai @pauloneves I can confirm that changing from `ckan/ckan-base:2.10.6` to `ckan/ckan-base:2.10.6-py3.10` (with root).

```
FROM ckan/ckan-base:2.10.6-py3.10
USER root
```

Many thanks for the hint!

**Comment by nickramser** [comment]
@kowh-ai, confirm switching to Debian base image worked for me as well with some tweaks to my Containerfile. Thanks for the info, I didn't realize the alpine images were deprecated.
